### DISCUSSION TITLE
[文章前缀无法生效] 升級至 4.x 之後, NEXT_PUBLIC_POST_URL_PREFIX 會導致頁面無法正常運作

### DISCUSSION THREAD

**Discussion by siygle** [tangly1024/NotionNext#1973] [Q&A]
**描述bug**
NEXT_PUBLIC_POST_URL_PREFIX 在升級至 4.x 之後無法正常運作。發生的主因應該是 Post 在 Next.js 新版本 page 的路由比對，在升版至 4.x 之後由原本的 `[page].js` 改為 `[prefix]/[slug].js`，當 NEXT_PUBLIC_POST_URL_PREFIX 為多層結構時，頁面無法正常顯示。

**复现步骤**
設定 NEXT_PUBLIC_POST_URL_PREFIX 的環境變數，數值為至少兩層以上。e.g., (%year%/%month%/%day%)

**期望的正常结果**
頁面可正常顯示。

**截图**
此為正常顯示的版本：
<img width="819" alt="image" src="https://github.com/tangly1024/NotionNext/assets/173408/94d057e9-1128-450a-8c4c-d87b24b3e634">

一併附上可重現錯誤的版本，這個[預覽版本](https://notion-next-4bxihtp1m-siygle.vercel.app/)是有問題的狀態，所有文章連結點擊之後都沒辦法正常顯示內文。

**环境**

- 操作系统: macOS
- 浏览器: chrome
- NotionNext版本: 4.0.11
- 主题: nobelium

**补充说明**
原本想說可以改成 [Catch-all segements](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes#optional-catch-all-segments) 不過它好像沒辦法搭配 [slug].js 使用。目前沒想法要怎麼修改，先回報這個問題。 🙇 
> Error: Catch-all must be the last part of the URL.

**Comment by tangly1024** [comment]
感谢反馈，这个问题影响比较大，我会关注一下

**Comment by siygle** [comment]
抱歉想詢問一下，這個問題是不是沒辦法在短時間之內解決了呢？
